plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,286 @@
+    this.versionRenamers_ =
+        Renamer.calculateRenamings(database, currVersion, newVersion);
+  }
+
+  /**
+   * Turns a database of renamings into an intermediate format that is
+   * easier to apply.
+   * @param {!Object} database The database of renames.
+   * @param {string} currVersion The version to migrate from.
+   * @param {string} newVersion The version to migrate to.
+   * @return {!Array<!VersionRenamer>} The collection of renamings to perform.
+   */
+  static calculateRenamings(database, currVersion, newVersion) {
+    currVersion = Renamer.coerce(currVersion);
+    newVersion = Renamer.coerce(newVersion);
+    const versions = Object.keys(database).sort(Renamer.compare);
+    const renamers /** !Array<!VersionRenamer> */ = [];
+    for (const version of versions) {
+      // Only process versions in the range (currVersion, ^newVersion].
+      if (Renamer.lte(version, currVersion)) continue;
+      if (Renamer.gt(version, newVersion)) break;
+
+      renamers.push(new VersionRenamer(database[version]));
+    }
+    return renamers;
+  }
+
+  /**
+   * Coerces the given string into a valid version (semver compliant or
+   * develop).
+   * @param {string} version  The version to coerce.
+   * @return  {string} The coerced version.
+   */
+  static coerce(version) {
+    return version === DEV_VERSION ?
+        version : semver.coerce(version).toString();
+  }
+
+  /**
+   * Compares the given versions. Compatible with Array.sort.
+   * @param {string} v1 The first version to compare.
+   * @param {string} v2 The second version to compare.
+   * @return {number} A number indicating the relationship between the versions.
+   */
+  static compare(v1, v2) {
+    if (v2 === DEV_VERSION) return -1;
+    if (v1 === DEV_VERSION) return 1;
+    return semver.compare(v1, v2);
+  }
+
+  /**
+   * Returns true if the first version is less than or equal to the second
+   * version.
+   * @param {string} v1 The version to compare.
+   * @param {string} v2 The version to compare against.
+   * @return {boolean} True if the first version is less than or equal to the
+   *     second one.
+   */
+  static lte(v1, v2) {
+    if (v2 === DEV_VERSION) return true;
+    if (v1 === DEV_VERSION) return false;
+    return semver.lte(v1, v2);
+  }
+
+  /**
+   * Returns true if the first version is greater than the second version.
+   * @param {string} v1 The version to compare.
+   * @param {string} v2 The version to compare against.
+   * @return {boolean} True if the first version is greater than the second one.
+   */
+  static gt(v1, v2) {
+    if (v2 === DEV_VERSION) return false;
+    if (v1 === DEV_VERSION) return true;
+    return semver.gtr(v1, `^${v2}`);
+  }
+
+  /**
+   * Applies the given renamings directly to a JavaScript string
+   * (presumably the contents of a developer's file).
+   * @param {string} str The string to apply the renamings in.
+   * @return {string} The file with renamings applied.
+   */
+  rename(str) {
```

Comment:
I'd suggest keeping this together with `calculateRenamings`, to make it easier to read both at the same time.